--- a/drivers/of/Kconfig
+++ b/drivers/of/Kconfig
@@ -108,14 +108,6 @@ config OF_UNITTEST
 	  While this option is selected automatically when needed, you can
 	  enable it manually to improve device tree unit test coverage.
 
-config OF_DTBO
-	bool "Device Tree DTBO"
-	select OF_DYNAMIC
-	select OF_FLATTREE
-	select OF_RESOLVE
-	help
-	  Device Tree DTBO
-
 config OF_NUMA
 	bool
 
@@ -123,4 +115,11 @@ config OF_DMA_DEFAULT_COHERENT
 	# arches should select this if DMA is coherent by default for OF devices
 	bool
 
+config OF_CONFIGFS
+	bool "Device Tree Overlay ConfigFS interface"
+	select CONFIGFS_FS
+	select OF_OVERLAY
+	help
+	  Enable a simple user-space driven DT overlay interface.
+
 endif # OF