diff --git a/meson.build b/meson.build
index ab687d8..1234567 100644
--- a/meson.build
+++ b/meson.build
@@ -3,6 +3,7 @@ project(
   ['c', 'cpp'],
   version : '8.1.0',
   meson_version : '>=0.50.0',
+  default_options : ['cpp_std=c++17'],
 )
 
 pkgconfig = import('pkgconfig')
@@ -76,6 +78,8 @@ add_project_arguments(
    '-DADB_VERSION="' + meson.project_version() + '"',
    '-DALLOW_ADBD_NO_AUTH=1',
-   '-Wno-unused-result'],
+   '-Wno-unused-result',
+   '-Wno-error=missing-include-dirs',
+   '-D_GNU_SOURCE'],
   language: 'cpp')
 
 executable(
