diff -u a/adb/sysdeps/posix/network.cpp b/adb/sysdeps/posix/network.cpp
--- a/adb/sysdeps/posix/network.cpp   2025-09-23 01:18:53.034934886 -0000
+++ b/adb/sysdeps/posix/network.cpp   2025-09-23 01:19:21.292826630 -0000
@@ -21,6 +21,7 @@
 #include <sys/socket.h>
 
 #include <string>
+#include <cstring>
 
 #include "adb_unique_fd.h"
 

diff -u a/base/logging.cpp b/base/logging.cpp
--- a/base/logging.cpp      2025-09-23 01:26:02.134693713 -0000
+++ b/base/logging.cpp      2025-09-23 01:27:30.304251308 -0000
@@ -40,6 +40,7 @@
 #include <mutex>
 #include <sstream>
 #include <string>
+#include <cstring>
 #include <utility>
 #include <vector>
 
diff -u a/base/file.cpp b/base/file.cpp
--- a/base/file.cpp         2025-09-23 01:31:39.130813168 -0000
+++ b/base/file.cpp         2025-09-23 01:32:02.000221147 -0000
@@ -26,6 +26,7 @@
 #include <memory>
 #include <mutex>
 #include <string>
+#include <cstring>
 #include <vector>
 
 #include "android-base/logging.h"
diff '--color=auto' -u -r a/adb/adb.cpp b/adb/adb.cpp
--- a/adb/adb.cpp       2025-09-23 01:34:19.164806704 -0000
+++ b/adb/adb.cpp       2025-09-23 01:37:39.741051978 -0000
@@ -34,6 +34,7 @@
 #include <condition_variable>
 #include <mutex>
 #include <string>
+#include <cstring>
 #include <thread>
 #include <vector>
 
diff '--color=auto' -u -r a/adb/adb_auth_host.cpp b/adb/adb_auth_host.cpp
--- a/adb/adb_auth_host.cpp     2025-09-23 01:34:19.163806698 -0000
+++ b/adb/adb_auth_host.cpp     2025-09-23 01:37:39.686351806 -0000
@@ -28,6 +28,7 @@
 #include <mutex>
 #include <set>
 #include <string>
+#include <cstring>
 
 #include <android-base/errors.h>
 #include <android-base/file.h>
diff '--color=auto' -u -r a/adb/adb_client.cpp b/adb/adb_client.cpp
--- a/adb/adb_client.cpp        2025-09-23 01:34:19.152806629 -0000
+++ b/adb/adb_client.cpp        2025-09-23 01:37:39.258048979 -0000
@@ -32,6 +32,7 @@
 #include <condition_variable>
 #include <mutex>
 #include <string>
+#include <cstring>
 #include <thread>
 #include <vector>
 
diff '--color=auto' -u -r a/adb/adb_io_test.cpp b/adb/adb_io_test.cpp
--- a/adb/adb_io_test.cpp       2025-09-23 01:34:19.152806629 -0000
+++ b/adb/adb_io_test.cpp       2025-09-23 01:37:39.287989846 -0000
@@ -26,6 +26,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #include <android-base/file.h>
 #include <android-base/test_utils.h>
diff '--color=auto' -u -r a/adb/adb_utils_test.cpp b/adb/adb_utils_test.cpp
--- a/adb/adb_utils_test.cpp    2025-09-23 01:34:19.153806635 -0000
+++ b/adb/adb_utils_test.cpp    2025-09-23 01:37:39.332835143 -0000
@@ -22,6 +22,7 @@
 #endif
 
 #include <string>
+#include <cstring>
 
 #include <gtest/gtest.h>
 
diff '--color=auto' -u -r a/adb/bugreport.cpp b/adb/bugreport.cpp
--- a/adb/bugreport.cpp 2025-09-23 01:34:19.153806635 -0000
+++ b/adb/bugreport.cpp 2025-09-23 01:37:39.344049513 -0000
@@ -19,6 +19,7 @@
 #include "bugreport.h"
 
 #include <string>
+#include <cstring>
 #include <vector>
 
 #include <android-base/file.h>
diff '--color=auto' -u -r a/adb/commandline.cpp b/adb/commandline.cpp
--- a/adb/commandline.cpp       2025-09-23 01:34:19.155937718 -0000
+++ b/adb/commandline.cpp       2025-09-23 01:37:39.401049867 -0000
@@ -33,6 +33,7 @@
 
 #include <memory>
 #include <string>
+#include <cstring>
 #include <thread>
 #include <vector>
 
diff '--color=auto' -u -r a/adb/file_sync_client.cpp b/adb/file_sync_client.cpp
--- a/adb/file_sync_client.cpp  2025-09-23 01:34:19.157806660 -0000
+++ b/adb/file_sync_client.cpp  2025-09-23 01:37:39.465448756 -0000
@@ -31,6 +31,7 @@
 #include <memory>
 #include <sstream>
 #include <string>
+#include <cstring>
 #include <vector>
 
 #include "sysdeps.h"
diff '--color=auto' -u -r a/adb/remount_service.cpp b/adb/remount_service.cpp
--- a/adb/remount_service.cpp   2025-09-23 01:34:19.163806698 -0000
+++ b/adb/remount_service.cpp   2025-09-23 01:37:39.710464976 -0000
@@ -28,6 +28,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #include <android-base/properties.h>
 
diff '--color=auto' -u -r a/adb/shell_service.cpp b/adb/shell_service.cpp
--- a/adb/shell_service.cpp     2025-09-23 01:34:19.164806704 -0000
+++ b/adb/shell_service.cpp     2025-09-23 01:37:39.765381496 -0000
@@ -90,6 +90,7 @@
 
 #include <memory>
 #include <string>
+#include <cstring>
 #include <thread>
 #include <unordered_map>
 #include <vector>
diff '--color=auto' -u -r a/adb/shell_service_test.cpp b/adb/shell_service_test.cpp
--- a/adb/shell_service_test.cpp        2025-09-23 01:34:19.158806666 -0000
+++ b/adb/shell_service_test.cpp        2025-09-23 01:37:39.511447644 -0000
@@ -21,6 +21,7 @@
 #include <signal.h>
 
 #include <string>
+#include <cstring>
 #include <vector>
 
 #include <android-base/strings.h>
diff '--color=auto' -u -r a/adb/socket_test.cpp b/adb/socket_test.cpp
--- a/adb/socket_test.cpp       2025-09-23 01:34:19.159806673 -0000
+++ b/adb/socket_test.cpp       2025-09-23 01:37:39.547922067 -0000
@@ -22,6 +22,7 @@
 #include <limits>
 #include <queue>
 #include <string>
+#include <cstring>
 #include <thread>
 #include <vector>
 
diff '--color=auto' -u -r a/adb/sockets.cpp b/adb/sockets.cpp
--- a/adb/sockets.cpp   2025-09-23 01:34:19.163806698 -0000
+++ b/adb/sockets.cpp   2025-09-23 01:37:39.723051866 -0000
@@ -28,6 +28,7 @@
 #include <algorithm>
 #include <mutex>
 #include <string>
+#include <cstring>
 #include <vector>
 
 #ifdef __ANDROID__
diff '--color=auto' -u -r a/adb/sysdeps.h b/adb/sysdeps.h
--- a/adb/sysdeps.h     2025-09-23 01:34:19.163806698 -0000
+++ b/adb/sysdeps.h     2025-09-23 01:37:39.671088942 -0000
@@ -27,6 +27,7 @@
 #include <errno.h>
 
 #include <string>
+#include <cstring>
 #include <vector>
 
 // Include this before open/close/unlink are defined as macros below.
@@ -88,6 +89,7 @@
 
 #include <memory>   // unique_ptr
 #include <string>
+#include <cstring>
 
 #include "fdevent.h"
 
@@ -367,6 +369,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #define OS_PATH_SEPARATORS "/"
 #define OS_PATH_SEPARATOR '/'
diff '--color=auto' -u -r a/base/file_test.cpp b/base/file_test.cpp
--- a/base/file_test.cpp        2025-09-23 01:34:19.169328779 -0000
+++ b/base/file_test.cpp        2025-09-23 01:37:39.837149620 -0000
@@ -23,6 +23,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #include "android-base/test_utils.h"
 
diff '--color=auto' -u -r a/base/include/android-base/errors.h b/base/include/android-base/errors.h
--- a/base/include/android-base/errors.h        2025-09-23 01:34:19.170725037 -0000
+++ b/base/include/android-base/errors.h        2025-09-23 01:37:39.853052673 -0000
@@ -31,6 +31,7 @@
 #define ANDROID_BASE_ERRORS_H
 
 #include <string>
+#include <cstring>
 
 namespace android {
 namespace base {
diff '--color=auto' -u -r a/base/logging_test.cpp b/base/logging_test.cpp
--- a/base/logging_test.cpp     2025-09-23 01:34:19.171806747 -0000
+++ b/base/logging_test.cpp     2025-09-23 01:37:39.945248167 -0000
@@ -24,6 +24,7 @@
 
 #include <regex>
 #include <string>
+#include <cstring>
 
 #include "android-base/file.h"
 #include "android-base/stringprintf.h"
diff '--color=auto' -u -r a/base/stringprintf_test.cpp b/base/stringprintf_test.cpp
--- a/base/stringprintf_test.cpp        2025-09-23 01:34:19.173806760 -0000
+++ b/base/stringprintf_test.cpp        2025-09-23 01:37:39.992317990 -0000
@@ -19,6 +19,7 @@
 #include <gtest/gtest.h>
 
 #include <string>
+#include <cstring>
 
 TEST(StringPrintfTest, HexSizeT) {
   size_t size = 0x00107e59;
diff '--color=auto' -u -r a/base/strings.cpp b/base/strings.cpp
--- a/base/strings.cpp  2025-09-23 01:34:19.173806760 -0000
+++ b/base/strings.cpp  2025-09-23 01:37:40.003465249 -0000
@@ -20,6 +20,7 @@
 #include <string.h>
 
 #include <string>
+#include <cstring>
 #include <vector>
 
 namespace android {
diff '--color=auto' -u -r a/base/utf8.cpp b/base/utf8.cpp
--- a/base/utf8.cpp     2025-09-23 01:34:19.173806760 -0000
+++ b/base/utf8.cpp     2025-09-23 01:37:40.034002098 -0000
@@ -23,6 +23,7 @@
 
 #include <algorithm>
 #include <string>
+#include <cstring>
 
 #include "android-base/logging.h"
 
